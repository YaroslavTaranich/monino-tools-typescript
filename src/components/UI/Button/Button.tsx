import { IconDefinition } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { HTMLAttributes } from 'react'

import styles from './button.module.scss'

interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
  children: string
  variant?: 'primary' | 'secondary' | 'menu'
  icon?: IconDefinition
}

function Button({ variant, icon, ...props }: ButtonProps) {
  const classNames = [styles.button, styles[variant!]]

  return (
    <button className={classNames.join(' ')} type="button" {...props}>
      {icon && <FontAwesomeIcon icon={icon} className={styles.icon} />}
      {props.children}
    </button>
  )
}

Button.defaultProps = {
  variant: 'primary',
  icon: undefined,
}

export default Button
