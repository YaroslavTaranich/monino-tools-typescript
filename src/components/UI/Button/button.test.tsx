import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { faClose } from '@fortawesome/free-solid-svg-icons'

import Button from './Button'

describe('testing button', () => {
  test('Simple button without icon', async () => {
    render(<Button>Hello</Button>)

    const button = screen.getByRole('button')
    const icon = screen.queryByRole('img')

    expect(button).toHaveTextContent(/hello/i)
    expect(button).toBeInTheDocument()
    expect(icon).toBeNull()
    expect(button).toMatchSnapshot()
  })

  test('Simple button without icon', async () => {
    render(<Button icon={faClose}>Hello</Button>)

    const button = screen.getByRole('button')
    const icon = document.querySelector('[data-icon="xmark"]')

    expect(button).toHaveTextContent(/hello/i)
    expect(icon).toBeInTheDocument()
    expect(button).toMatchSnapshot()
  })
})
