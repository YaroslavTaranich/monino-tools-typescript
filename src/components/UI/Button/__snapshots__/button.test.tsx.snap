// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render mobile menu 1`] = `
<button
  class="button primary"
  type="button"
>
  Hello
</button>
`;

exports[`testing button Simple button without icon 1`] = `
<button
  class="button primary"
  type="button"
>
  Hello
</button>
`;

exports[`testing button Simple button without icon 2`] = `
<button
  class="button primary"
  type="button"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-xmark icon"
    data-icon="xmark"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 320 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
      fill="currentColor"
    />
  </svg>
  Hello
</button>
`;

exports[`testing button render mobile menu 1`] = `
<button
  class="button primary"
  type="button"
>
  Hello
</button>
`;
